table/tables/partition.go

```diff
@@ -146,13 +171,7 @@ func (t *PartitionedTable) locatePartition(ctx sessionctx.Context, pi *model.Par
 
 // GetPartition returns a Table, which is actually a Partition.
 func (t *PartitionedTable) GetPartition(pid int64) table.Table {
-	var ret Partition
-	// Make a shallow copy, change ID to partition ID.
-	ret.tableCommon = t.tableCommon
-	ret.partitionID = pid
-	ret.recordPrefix = tablecodec.GenTableRecordPrefix(pid)
-	ret.indexPrefix = tablecodec.GenTableIndexPrefix(pid)
-	return &ret
+	return t.partitions[pid]
```

Comment:
Why change here?

Reply:
`GetPartition` function to create a `Partition` and `Index` everytime it's called, this is bad for performance.
So I store those information when `PartitionedTable` is created, and just get it here.